! SOURCE
+ Article
- https://flaven.fr/2025/02/the-green-ai-myth-carbon-costs-of-ml-models-insights-on-geo-and-deepseek-fine-tuning/

+ Videos
- https://www.youtube.com/channel/UCnUBoVx9Yai3wirPBvNpNQw

+ Github
- https://github.com/bflaven/

+ Abstract
This post explores the multifaceted role of AI, focusing on its optimization (GEO), environmental impact, and corporate social responsibility. It discusses how AI's emergence is reshaping web searches and business models, potentially disrupting the link economy.  A significant portion examines AI's carbon footprint, referencing tools like CodeCarbon and Eco2AI for measurement, while advocating for a rational approach to AI usage and awareness of the "Jevons Paradox." The text also assesses DeepSeek, highlighting its potential to challenge existing AI ecosystems and offering insights into fine-tuning custom models with Python. Furthermore, it provides an interesting, and humorous, semantic study of the names of AI leader brands (DeepSeek, Mistral, Claude) and their cultural associations, and closes by presenting the German tradition of "Unwort des Jahres" (non-word of the year) in relation to AI.



+ Keywords
#AI #LLM #Mistral #Faiss #CLIP #ChatGPT #Perplexity #Claude #Grok #Streamlit #Innovation2025 

--- Intro
You can read the article on my blog.
The ‘Green’ AI Myth: Carbon Costs of ML Models & Insights on GEO and DeepSeek Fine-tuning
- https://wp.me/p3Vuhl-3nO



--- Audio version
This post is also an experiment to test NotebookLM. So, here is this regular blog post "The ‘Green’ AI Myth: Carbon Costs of ML Models & Insights on GEO and DeepSeek Fine-tuning" converted into a podcast using NotebookLM.


The Big Dive : The 'Green' AI Myth: Carbon Costs of ML Models & Insights on GEO and DeepSeek Fine-tuning
- https://on.soundcloud.com/PVzHx8TEb5b65HHg9


! VIDEOS

--- create the env with anaconda

"""
[env]
# Conda Environment
conda create --name ia_debunk python=3.9.13
conda create --name carbon_footprint python=3.9.13
conda info --envs
source activate ia_debunk
source activate carbon_footprint
conda deactivate


# BURN AFTER READING
source activate ia_debunk
source activate carbon_footprint

# if needed to remove
conda env remove -n [NAME_OF_THE_CONDA_ENVIRONMENT]
conda env remove -n ia_debunk
conda env remove -n carbon_footprint

# install packages
python -m pip install streamlit 
python -m pip install streamlit

# update conda 
conda update -n base -c defaults conda

# to export requirements
pip freeze > requirements.txt

# [path]
cd /Users/brunoflaven/Documents/03_git/ia_usages/ia_kpi_carbon_footprint_deepseek/ia_kpi_carbon_llm




# launch the file
python _api_call_demo.py

"""

+ 1. Generative Engine Optimization (GEO)

+ 2. IA, the ultimate Carbon Footprint
+ 3. Selecting Tools to find KPI for AI Carbon Footprint
--- 3.1 Using CodeCarbon
--- 3.2 Using Eco2AI
--- 3.3 Using carbontxt.org
--- 3.4 Other tools
+ 4. How-to analyse the RSE impact of IA
+ 5. Using DeepSeek
+ 6. DeepSeek Fine-Tuning: Revolutionizing Custom AI Model Development with Python

+ 1. ia_kpi_carbon_llm
--- ia_kpi_carbon_llm.mov

- DESC: measure carbon footprint for a local LLM like mistral name "mistral-poet", operated with Ollama.

cd /Users/brunoflaven/Documents/03_git/ia_usages/ia_kpi_carbon_footprint_deepseek/ia_kpi_carbon_llm

--- COMMAND How to add or remove LLM on ollama 

# show help
ollama help


# list models
ollama list

# run a model
ollama run mistral:latest

# remove a model
ollama rm mistral:latest

# idem
# I have created 
ollama run mistral-poet:latest
ollama rm mistral-poet:latest

--- some queries...
What is Mayonnaise?


--- COMMAND How to create mistral-poet:latest (using modelfile)
Check https://github.com/ollama/ollama/blob/main/docs/modelfile.md

# create model with modefile
cd /Users/brunoflaven/Documents/03_git/ia_usages/ia_kpi_carbon_footprint_deepseek/ia_kpi_carbon_llm/

# command to create a model based on mistral named mistral poet
ollama create mistral-poet -f Modelfile_mistral_poet


! CONSOLE_SCREEN (commands)

# GO TO DIR
cd /Users/brunoflaven/Documents/03_git/ia_usages/ia_kpi_carbon_footprint_deepseek/ia_kpi_carbon_llm

# LAUNCH THE ENV
conda info --envs
source activate carbon_footprint

# launch the file



+ webapp with streamlit
streamlit run 001_carbon_footprint_monitoring_system_streamlit.py
streamlit run 002_carbon_footprint_decision_matrix_implementation_streamlit.py
streamlit run 002a_carbon_footprint_decision_matrix_implementation_streamlit.py


+ carbon footprint using codecarbon
python 004_carbon_footprint_codecarbon_basic_usage.py
python 005_carbon_footprint_codecarbon_decorator_usage.py
python 006_carbon_footprint_codecarbon_task_manager_usage.py
python 007_carbon_footprint_codecarbon_site_explicit_object.py
python 008_carbon_footprint_codecarbon_site_context_manager.py
python 009_carbon_footprint_codecarbon_site_using_decorator.py
python 010_carbon_footprint_codecarbon_local_llm.py
python 012_carbon_footprint_codecarbon_local_llm.py

+ carbon footprint using eco2ai
python 014_carbon_footprint_eco2ai.py
python 015_carbon_footprint_eco2ai.py
python 016_carbon_footprint_eco2ai.py










